### PR TITLE
Define Gelu operation

### LINKED ISSUE #626: Need Gelu operation
**Use Case:**
`Gelu` performs the gaussian error linear unit (GELU) activation function: `f(x) = 0.5 * x * (1.0 + erf(x / sqrt(2)))`.
Some transformer models such as the [whisper base](https://huggingface.co/openai/whisper-base) contain decomposed small ops like `Add`, `Div`, `Erf`, `Mul` following the `MatMul` and `Conv` operations. These small ops can be composed as a higher level operation `Gelu`.
Furthermore, the `Gelu` can also be fused into the operations like  `MatMul` and `Conv` , it has been confirmed that the fusion can significantly improve the performance of a graph's execution at least on DirectML backend.

**Sample models:**
[whisper base](https://huggingface.co/openai/whisper-base)

**Cross-framework support:**
[ONNX](https://onnx.ai/onnx/operators/onnx__Gelu.html), [TensorFlow](https://www.tensorflow.org/api_docs/python/tf/nn/gelu), [PyTorch](https://pytorch.org/docs/stable/generated/torch.nn.GELU.html)

**Cross-platform implementability:**
[CoreML](https://apple.github.io/coremltools/source/coremltools.converters.mil.mil.ops.defs.html#coremltools.converters.mil.mil.ops.defs.iOS15.activation.gelu), [DirectML](https://learn.microsoft.com/en-us/windows/ai/directml/api/ns-directml-dml_activation_gelu_operator_desc), [OpenVINO](https://docs.openvino.ai/2023.3/openvino_docs_ops_activation_GELU_7.html), 

So I propose this new operation to WebNN Spec, my PR is WIP. @huningxin @fdwr

### PR DESCRIPTION
**Pull request by mingmingtasd**
To fix #626 

PTAL, thanks! @huningxin

***
<a href="https://pr-preview.s3.amazonaws.com/mingmingtasd/webnn/pull/628.html" title="Last updated on Apr 4, 2024, 1:13 AM UTC (58d76da)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/628/60b8e0a...mingmingtasd:58d76da.html" title="Last updated on Apr 4, 2024, 1:13 AM UTC (58d76da)">Diff</a>

### CODE REVIEW

**Review by huningxin — COMMENTED** [review]
Looks good with some nits. Thanks @mingmingtasd. #626 is on 4/4 WG meeting [agenda](https://github.com/webmachinelearning/meetings/blob/main/telcons/2024-04-04-wg-agenda.md#%EF%B8%8F-feature-request), let's wait for WG's decision.

**Comment by mingmingtasd** [comment]
> Looks good with some nits. Thanks @mingmingtasd. #626 is on 4/4 WG meeting [agenda](https://github.com/webmachinelearning/meetings/blob/main/telcons/2024-04-04-wg-agenda.md#%EF%B8%8F-feature-request), let's wait for WG's decision.

Got it, thanks!

**Comment by inexorabletash** [comment]
Patch LGTM, no coding style issues noted.

File: index.bs
Comment (fdwr) [inline index.bs:2835]:
```suggestion
    other operations as follows. However, user agents typically have a more
```

File: index.bs
Comment (fdwr) [inline index.bs:2836]:
```suggestion
    efficient implementation for it. Therefore its usage is encouraged from the
```

**Review by fdwr — COMMENTED** [review]
Thank you Mingming for creating this. I only have some wording polish, and so I'll go ahead and just commit them. I guess we should wait to actually *Approve* it after we talk in W3C meeting?

**Comment by anssiko** [comment]
Discussed on [WebML WG Teleconference – 4 April 2024](https://www.w3.org/2024/04/04-webmachinelearning-minutes.html#t13).

This new op passes our [proposing and adding a new op](https://github.com/webmachinelearning/webnn/blob/main/CONTRIBUTING.md#proposing-and-adding-a-new-operation) tests with results documented in https://github.com/webmachinelearning/webnn/issues/626#issue-2212579157

When all the reviews comments have been addressed, I consider this ready to be merged.

**Review by fdwr — APPROVED** [review]

**Review by huningxin — APPROVED** [review]
LGTM!

**Pull request merged**
